### PR TITLE
Card Creation

### LINKED ISSUE #11: Card creation functionality
- [x] Create card from a `+` button
- [x] Add card directly to column

### PR DESCRIPTION
**Pull request by Trinityyi**
Resolves #11

### CODE REVIEW

**Comment by Trinityyi** [comment]
@Chalarangelo could use some styles

File: src/components/AddButton.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  useClickOutside(clickRef, () => {
+    if (isEditable) {
+      onSubmit(clickRef.current.querySelector(`#${id}`).value);
+      setIsEditable(false);
+    }
+  });
+
+  return (
+    <div className={combineClassNames`add-button-wrapper ${wrapperClassName}`} ref={clickRef}>
+      {
```
Comment (Chalarangelo) [inline src/components/AddButton.jsx:27]: [minor] Let's inline the condition, so there's one indentation level less here.

File: src/components/AddButton.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        isEditable ? (
+          <input
+            className={inputClassName ? inputClassName : undefined}
```
Comment (Chalarangelo) [inline src/components/AddButton.jsx:30]: Prefer `null` instead:
```suggestion
            className={inputClassName ? inputClassName : null}
```
If I remember correctly either one will result in an element without a `className` as React drops either one.

File: src/components/Column.jsx [hunk 3]
```diff
@@ -13,19 +20,35 @@ const Column = ({
...
+          <Card key={card.id} card={card}/>
         ))}
       </ul>
+      <AddButton
+        id = {`btn-${id}`}
+        name = {`btn-${id}`}
+        onSubmit ={cardTitle => createCard({title: cardTitle}, id)}
+        buttonText ="Add a card"
+        wrapperClassName="column-action-add"
+        buttonClassName = "btn-column-add"
```
Comment (Chalarangelo) [inline src/components/Column.jsx:32]: [formatting] :scissors: the spaces around the `=`, try to be consistent with your formatting. :wink:

File: src/components/MainArea.jsx [hunk 4]
```diff
@@ -29,6 +29,7 @@ const MainArea = ({
       <div className="main-area-content">
         {Object.values(columns).map(col => (
           <Column
+            id={col.id}
```
Comment (Chalarangelo) [inline src/components/MainArea.jsx:32]: We might want to desctructure `col` in this loop at some point.

File: src/components/AddButton.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import useClickOutside from '../hooks/useClickOutside';
+import combineClassNames from '@chalarangelo/combine-class-names';
+
+const AddButton = ({
```
Comment (Chalarangelo) [inline src/components/AddButton.jsx:6]: TODO: Add specs and docs.

**Pull request merged**